osf/models/registrations.py

```diff
@@ -296,7 +350,7 @@ def embargo_registration(self, user, end_date, for_existing_registration=False,
         :raises: PermissionsError if user is not an admin for the Node
         :raises: ValidationError if end_date is not within time constraints
         """
-        if not self.is_admin_contributor(user):
+        if not self.is_admin_contributor(user) and not self.provider.get_group('moderator').user_set.filter(id=user.id).exists():
```

Comment:
Why here and not elsewhere? And is there an issue with Moderator privileges accidentally appearing to also provide admin privileges?

Reply:
I wrote this in because I was hitting permissions errors when the moderator did the final approval, as the moderator is passed in to this method as the user.